main.c

```diff
@@ -239,6 +253,8 @@ static void apply_profile(struct kanshi_state *state,
 	}
 
 	zwlr_output_configuration_v1_apply(config);
+
+	wl_display_roundtrip(state->display);
```

Comment:
Why is this necessary?

Reply:
Okay so this is a bit confusing why it needed to be in this specific spot, but the roundtrip is because the response to the reload IPC message is not really useful unless it's sent after the new output configuration gets applied, otherwise there is a race condition. Demonstrating with a shell script that detects new monitors and then tries to automatically configure them and set a background or something like that:

```bash
outputs=$(swaymsg -rt get_outputs)
# ... process the outputs and detect new ones, generate a config ...
echo $myconfig >> ~/.config/kanshi/config
kanshictl reload
swaybg $bgconfig
```

If `kanshictl reload` returns immediately without waiting for this roundtrip, then swaybg could potentially run before the resolution changes. Note that this works in sway (and wayfire I think) because it applies the output changes immediately. But I suppose other compositors could wait a frame, or be multi-threaded, or pop up a dialog to confirm the changes first... all of which would also cause this type of script to fail. The workaround in that case (or just without the roundtrip) would be to repeatedly call `swaymsg -t get_outputs` in a while loop after kanshictl, to confirm the resolution changes, which is not great.

Reply:
Actually it just occurred to me that this can be replaced with a wl_display_sync.